src/main/java/de/zalando/aruha/nakadi/config/NakadiConfig.java

```diff
@@ -86,18 +79,36 @@ public ZooKeeperHolder zooKeeperHolder() {
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaLocationManager getKafkaLocationManager() {
         return new KafkaLocationManager();
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaFactory kafkaFactory() {
         return new KafkaFactory(getKafkaLocationManager());
     }
 
     @Bean
-    public KafkaRepositorySettings topicRepositorySettings() {
+    @Profile("kafka")
+    public KafkaRepositorySettings kafkaRepositorySettings() {
         return new KafkaRepositorySettings();
     }
 
+    @Bean
+    public TopicRepository topicRepository() {
+        if (environment.acceptsProfiles("kafka")) {
```

Comment:
it'd be simpler now not having any of these profiles for this.
a) once we support other brokers nothing is telling us that the same nakadi instance will not be able to use more than one 
b) having this now forces us to specify this profile, which is not adding any value.